org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchingUtils.java

```diff
@@ -39,6 +40,9 @@ public class JavaLaunchingUtils {
 	public static File createFileForArgument(String timeStamp, File processTempFilesDir, String launchConfigName, String optionalString) {
 		String child = String.format(org.eclipse.jdt.internal.launching.LaunchingPlugin.LAUNCH_TEMP_FILE_PREFIX
 				+ optionalString, launchConfigName, timeStamp);
+		if (child.length() > MAX_LAUNCH_CONFIG_NAME_LENGTH) {
```

Comment:
We don't have to check the length of child, we have to check the length of launchConfigName before using it for creating child.